hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -227,6 +233,8 @@ public AzureBlobFileSystemStore(
     final String fileSystemName = authorityParts[0];
     final String accountName = authorityParts[1];
     this.fsBackRef = abfsStoreBuilder.fsBackRef;
+    this.fsCreateCallback = abfsStoreBuilder.fsCreateCallback;
```

Comment:
Do we really need these implementations to flow all the way from file system to store to client to rename/delete handlers??

These handlers can call them directly on client??

Reply:
These callbacks can be used to create files, open a file, which will point to the methods of store, so that the fallbacks in append flow can be taken care of.

Why not call client method directly?
Since we are having multiple fallbacks in append flow, if the caller would call client method, it would also have to take care of those fallbacks. These callbacks would help the caller get required behavior without any explicit fallback handling.

Reply:
These callbacks have been removed. RenameAtomicity now have the intellegence of calling client methods to get the required work done. Thanks!